### ISSUE TITLE
FaceX Library 

### ISSUE THREAD

**Issue by AdityaNikhil** [akshitagupta15june/Face-X#312]
## About ❓
A unified library for **FaceX** to run all the FaceX algorithms using only one line of code. 

### Running cartoonify using FaceX library
    import facex 
    import cv2
    
    img = facex.cartoonify('your-img.jpg', method='opencv')
    cv2.imshow(img)
    cv2.waitkey()

Similarly we can run,

    facex.face_detect('your-img.jpg', method='opencv') #Face detection
    facex.face_mask('your-img.jpg', method='opencv')   #Face mask detection
    
    And many more....

## Pending Tasks ⏰
1) Release facex library V1.0 [[PR]](https://github.com/akshitagupta15june/Face-X/pull/313)
2) Refine the environment for easy access to the algorithms.
3) Make a **facex pip package**. 
4) Make clear documentation to use the package.
5) Make clear documentation for the contributors to link the algorithms with the package. 
6) Add more algorithms.

## Contributions are welcome 🤗
Feel free to suggest any changes or fixes for the benefit of the package by commenting down below.

## Notice to the maintainer
1) Please pin this issue.  
2) Please add **FaceX-API** label to track activities regarding this API.

**Comment by AdityaNikhil** [comment]
@akshitagupta15june  Please don't close this issue. Please read the **Contributions are welcome** ,**pending tasks** and **Notice to the maintainer** sections. I'd like to keep this open so to welcome more contributions and keep track of the progress of this library.

**Comment by akshitagupta15june** [comment]
Sure i have opened the issue

**Comment by AdityaNikhil** [comment]
Please pin this issue and add **FaceX-API** label.

**Comment by Hamza3226** [comment]
Hello, I am a GSSOC'21 participant and would like to contribute to this project.

**Comment by animesh-007** [comment]
> Please pin this issue and add **FaceX-API** label.

This is a good issue. @AdityaNikhil will it be okay if I contribute to this?

**Comment by AdityaNikhil** [comment]
@Hamza3226  @animesh-007 With pleasure✌
Check the **pending tasks** section in the above issue. See if you can contribute to any of those or suggest new ideas by commenting here. Would love to collaborate!

**Comment by akshitagupta15june** [comment]
sure

**Comment by akshitagupta15june** [comment]
@AdityaNikhil can you guide the participants in this issue.

**Comment by AdityaNikhil** [comment]
@akshitagupta15june Of course..!

**Comment by akshitagupta15june** [comment]
@Hamza3226 and @animesh-007

**Comment by Hamza3226** [comment]
Do we have to cartoonify an image ?

**Comment by AdityaNikhil** [comment]
@Hamza3226  That's been done..!
Please check the **How to use** section [here](https://github.com/akshitagupta15june/Face-X/tree/master/facex#how-to-use) and get familiar with the codebase here. 
@akshitagupta15june Can you open a discussion section for this repo. It'll be easier to make discussions.

**Comment by akshitagupta15june** [comment]
done @AdityaNikhil

**Comment by subhangi2731** [comment]
@akshitagupta15june if this issue is open can I please work on this

**Comment by AdityaNikhil** [comment]
@subhangi2731 
Of course, you can work on this issue. It's been a while since I worked on this.
You can check the homepage of this library [here](https://github.com/akshitagupta15june/Face-X/blob/master/facex-library). Let me know if you like to discuss anything.

**Comment by nozwock** [comment]
I'm surprised this hasn't been done already. If you'd like, I can take on the task. Creating a pip package shouldn't take more than a few minutes. Regarding the specifics of what this library should include, please provide me with the details.

**Comment by nozwock** [comment]
I'm also unsure why the `facex-library/GCA-Gender Classification API` directory is located within the `facex-lib` package. It doesn't appear to be relevant or connected to the library.
